python/katana/example_data.py

```diff
@@ -105,17 +106,25 @@ def _get_dataset(rel_path):
     since we might have downloaded the datasets, and our cache could be invalid
     try re-downloading the datasets if we are unable to locate rel_path
     """
-    try:
-        datasets_path = _get_test_datasets_directory()
-    except ValueError:
-        # try to re-download the datasets
-        datasets_path = _get_test_datasets_directory(invalidate_cache=True)
-
+    max_attempts = 5
+    count = 0
+    invalidate_cache = False
+    datasets_path = _get_test_datasets_directory(invalidate_cache=invalidate_cache)
     path = datasets_path / rel_path
-    if not path.exists():
-        raise ValueError(f"Dataset does not exist at {path}")
+    while not path.exists() or count < max_attempts:
+        try:
+            # lame backoff, costs nothing on first attempt
+            time.sleep(count)
+            datasets_path = _get_test_datasets_directory(invalidate_cache=invalidate_cache)
+            path = datasets_path / rel_path
+            return path
+        except ValueError as e:
+            print(f"Observed {e} while getting test datasets, ignoring and trying again")
```

Comment:
Use `logging`. https://docs.python.org/3/howto/logging.html#logging-basic-tutorial
Please use the `logger = logging.getLogger(__name__)` convention.